### PR TITLE
Fix #263: update to unicode 8.0.0. Also remove esutils.

### LINKED ISSUE #263: upgrade to Unicode 8.0.0
This will include updating our ID_Start, ID_Continue, and Whitespace regexes. See https://github.com/tc39/ecma262/pull/300 for more info.

### PR DESCRIPTION
**Pull request by bakkot**
esutils has not published a version with unicode 8.0.0, so replace the parts of it we need with in-house versions of the required methods.

### CODE REVIEW

File: scripts/generate-unicode-data.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+let regenerate = require('regenerate');
+let unicodeVersion = '8.0.0';
+
+let ascii = c => c < 128;
+let nonAscii = c => c >= 128;
+
+let makeCharRegex = s => `/^${regenerate().add(...s).toString()}$/`;
+let makeCharArray = s => JSON.stringify(regenerate().add(...s).toArray());
+
+let whitespace = [0x09, 0x0B, 0x0c, 0x20, 0xA0, 0xFEFF].concat(require(`unicode-${unicodeVersion}/General_Category/Space_Separator/code-points`));
```
Comment (michaelficarra) [inline scripts/generate-unicode-data.js:12]: What's the point of including these when we're just going to filter them out?

File: scripts/generate-unicode-data.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+let whitespaceArray = makeCharArray(whitespace.filter(a => a >= 256));
+
+
+let lineTerminators = [0x0A, 0x0D, 0x2028, 0x2029];
```
Comment (michaelficarra) [inline scripts/generate-unicode-data.js]: This appears unused.

**Review by michaelficarra — COMMENTED** [review]
Are we okay using Set and Array#fill? If we're using Set, would checking set inclusion be faster than the regexp?

**Comment by bakkot** [comment]
> Are we okay using Set and Array#fill? If we're using Set, would checking set inclusion be faster than the regexp?

We are (node 4, woo!), but it's not faster. I checked.

File: scripts/generate-unicode-data.js (hunk 1 above)
Comment (bakkot) [reply scripts/generate-unicode-data.js:12]: Eh... the idea was that this is all of the whitespace. But sure, I can remove them.

File: scripts/generate-unicode-data.js (hunk 2 above)
Comment (bakkot) [reply scripts/generate-unicode-data.js]: Whoops. Leftover code from a previous iteration.

File: scripts/generate-unicode-data.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+let idStartBool = Array(128).fill(0).map((v, i) => idStartSmall.indexOf(i) !== -1);
+
+
+let idContinue = ['$'.charCodeAt(0), '_'.charCodeAt(0), 0x200C, 0x200D].concat(require(`unicode-${unicodeVersion}/Binary_Property/ID_Continue/code-points`));
+let idContinueSmall = idContinue.filter(ascii);
+let idContinueLarge = idContinue.filter(nonAscii);
+let idContinueBool = Array(128).fill(0).map((v, i) => idContinueSmall.indexOf(i) !== -1);
+
+
+console.log(`// Generated by scripts/generate-unicode-data.js
```
Comment (michaelficarra) [inline scripts/generate-unicode-data.js]: Generate this name from `__filename`?

**Comment by bakkot** [comment]
@michaelficarra addressed comments.

File: src/utils.js [hunk 4]
```diff
@@ -14,23 +14,81 @@
...
+  'instanceof',
+]);
 
 export function isStrictModeReservedWord(id) {
-  return isReservedWordES6(id, true);
+  return strictReservedWords.has(id);
+}
+
+export function isWhiteSpace(ch) {
+  return ch === 0x20 || ch === 0x09 || ch === 0x0B || ch === 0x0C || ch === 0xA0 || ch > 0x167F && whitespaceArray.indexOf(ch) !== -1;
```
Comment (michaelficarra) [inline src/utils.js]: I would prefer that this function mirrored `isIdentifierStart`.

**Review by michaelficarra — CHANGES_REQUESTED** [review]

File: src/utils.js (hunk 4 above)
Comment (bakkot) [reply src/utils.js]: Done, kind of.

**Review by michaelficarra — APPROVED** [review]

**Pull request merged**